modules/vault-cluster/main.tf

```diff
@@ -1,52 +1,73 @@
 
 resource "aws_autoscaling_group" "autoscaling_group" {
-  name_prefix = "${var.cluster_name}"
+  name_prefix = var.cluster_name
 
-  launch_configuration = "${aws_launch_configuration.launch_configuration.name}"
+  launch_configuration = aws_launch_configuration.launch_configuration.name
 
-  availability_zones  = ["${var.availability_zones}"]
-  vpc_zone_identifier = ["${var.subnet_ids}"]
+  availability_zones  = var.availability_zones
+  vpc_zone_identifier = var.subnet_ids
 
   # Use a fixed-size cluster
-  min_size             = "${var.cluster_size}"
-  max_size             = "${var.cluster_size}"
-  desired_capacity     = "${var.cluster_size}"
-  termination_policies = ["${var.termination_policies}"]
+  min_size             = var.cluster_size
+  max_size             = var.cluster_size
+  desired_capacity     = var.cluster_size
+  termination_policies = [var.termination_policies]
 
-  health_check_type         = "${var.health_check_type}"
-  health_check_grace_period = "${var.health_check_grace_period}"
-  wait_for_capacity_timeout = "${var.wait_for_capacity_timeout}"
+  health_check_type         = var.health_check_type
+  health_check_grace_period = var.health_check_grace_period
+  wait_for_capacity_timeout = var.wait_for_capacity_timeout
 
-  enabled_metrics = ["${var.enabled_metrics}"]
+  enabled_metrics = var.enabled_metrics
 
   # Use bucket and policies names in tags for depending on them when they are there
   # And only create the cluster after S3 bucket and policies exist
   # Otherwise Vault might boot and not find the bucket or not yet have the necessary permissions
   # Not using `depends_on` because these resources might not exist
-  tags = ["${concat(
-    list(
-      map(
-        "key", var.cluster_tag_key,
-        "value", var.cluster_name,
-        "propagate_at_launch", true,
-        "using_s3_bucket_backend",  element(concat(aws_iam_role_policy.vault_s3.*.name, list("")), 0),
-        "s3_bucket_id", element(concat(aws_s3_bucket.vault_storage.*.id, list("")), 0),
-        "using_auto_unseal",  element(concat(aws_iam_role_policy.vault_auto_unseal_kms.*.name, list("")), 0),
-      )
-    ),
-    var.cluster_extra_tags)
-  }"]
+  tag {
+    key                 = var.cluster_tag_key
+    value               = var.cluster_name
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "using_s3_bucket_backend"
+    value               = element(concat(aws_iam_role_policy.vault_s3.*.name, [""]), 0)
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "s3_bucket_id"
+    value               = element(concat(aws_s3_bucket.vault_storage.*.id, [""]), 0)
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "using_auto_unseal"
+    value               = element(concat(aws_iam_role_policy.vault_auto_unseal_kms.*.name, [""]), 0)
+    propagate_at_launch = true
+  }
+
+  dynamic "tag" {
```

Comment:
Had to modify this to keep my module invocation the same. Does that seem correct?

Sample code:
```  
cluster_extra_tags = [
    {
      key                 = "Environment"
      value               = var.environment
      propagate_at_launch = true
    }
  ]
```

My modification:
```  
dynamic "tag" {
    for_each = [for s in var.cluster_extra_tags: {
      key                 = s.key
      value               = s.value
      propagate_at_launch = s.propagate_at_launch
    }]

    content {
      key                 = tag.value.key
      value               = tag.value.value
      propagate_at_launch = tag.value.propagate_at_launch
    }
  }
```